sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/WebExceptionRetryPolicy.java

```diff
@@ -3,28 +3,32 @@
 
 package com.azure.cosmos.implementation;
 
+import com.azure.cosmos.CosmosException;
 import com.azure.cosmos.implementation.apachecommons.lang.time.StopWatch;
 import com.azure.cosmos.implementation.directconnectivity.WebExceptionUtility;
+import com.azure.cosmos.implementation.http.HttpTimeoutPolicy;
+import com.azure.cosmos.implementation.http.HttpTimeoutPolicyDefault;
+import com.azure.cosmos.implementation.http.ResponseTimeoutAndDelays;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import reactor.core.publisher.Mono;
 
+import java.net.URI;
 import java.time.Duration;
 import java.util.concurrent.TimeUnit;
 
 public class WebExceptionRetryPolicy implements IRetryPolicy {
     private final static Logger logger = LoggerFactory.getLogger(WebExceptionRetryPolicy.class);
 
-    // total wait time in seconds to retry. should be max of primary reconfigrations/replication wait duration etc
-    private final static int waitTimeInSeconds = 30;
-    private final static int initialBackoffSeconds = 1;
-    private final static int backoffMultiplier = 2;
-
     private StopWatch durationTimer = new StopWatch();
-    private int attemptCount = 1;
-    // Don't penalise first retry with delay.
-    private int currentBackoffSeconds = WebExceptionRetryPolicy.initialBackoffSeconds;
     private RetryContext retryContext;
+    private int retryDelay;
+    private RxDocumentServiceRequest request;
+    private HttpTimeoutPolicy timeoutPolicy;
+    private boolean isReadRequest;
+    private int retryCount = 0;
+    private URI locationEndpoint;
+    private boolean isOutOfRetries;
```

Comment:
Curious if `retryCount` & `retryDelay` can be leverage `AtomicInteger`.

Reply:
Yeah, we can but it won't make any specific improvement.